src/net.h

```diff
@@ -113,6 +113,17 @@ struct CSerializedNetMsg
     std::string m_type;
 };
 
+/** Different types of connections to a peer. This enum encapsulates the
+ * information we have available at the time of opening or accepting the
+ * connection. Aside from INBOUND, all types are initiated by us. */
+enum class ConnectionType {
+    INBOUND, /**< peer initiated connections */
+    OUTBOUND, /**< full relay connections (blocks, addrs, txns) made automatically. Addresses selected from AddrMan. */
```

Comment:
I like the idea of a connection type enum, but I'd like to rename `INBOUND` and `OUTBOUND` here as it's a bit of a strange duck (as commented earlier, it's not the exclusive thing identifying this type of connection). Maybe `FULL_RELAY_INBOUND` and `FULL_RELAY_OUTBOUND`?

Reply:
with `INBOUND` I do think this is the only known characteristic of the connection, but I'm happy to rename `OUTBOUND` to `FULL_RELAY_OUTBOUND`.

Reply:
Sounds good to me!

Reply:
renamed `OUTBOUND` to `OUTBOUND_FULL_RELAY` in PR #19724 here: 34135a51775dc72809097cfb90b8611eee00106e